Bottom: f089a8416c9dd82c0d138e31743ef84f25c4bd11
Top:    3c8dd62fabab6eb55182688e7094a5fef8a095a0
Author: Mark de Wever <koraq@xs4all.nl>
Date:   2021-08-07 16:18:32 +0200

Refresh of add-a-calculator-controller

---

diff --git a/src/modules/calculator/controller.cpp b/src/modules/calculator/controller.cpp
index 3f3c12d..793ebde 100644
--- a/src/modules/calculator/controller.cpp
+++ b/src/modules/calculator/controller.cpp
@@ -91,10 +91,10 @@ void tcontroller::append(std::string_view data) noexcept {
 void tcontroller::push() noexcept {
   try {
     push(model_.input_steal());
+    model_.diagnostics_clear();
   } catch (const std::exception &e) {
     diagnostics_set(e);
   }
-  model_.input_clear();
 }
 
 void tcontroller::math_add() noexcept {
